compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -749,6 +749,22 @@ fn infer_placeholder_type(
     span: Span,
     item_ident: Ident,
 ) -> Ty<'_> {
+    let contains_anonymous = |ty: Ty<'_>| -> bool {
+        let mut contains_anonymous = false;
+        for gen_arg in ty.walk() {
+            if let ty::subst::GenericArgKind::Type(inner_ty) = gen_arg.unpack() {
+                match inner_ty.kind() {
+                    ty::FnDef(..) | ty::Closure(..) | ty::Generator(..) => {
+                        contains_anonymous = true;
+                        break;
+                    }
+                    _ => {}
+                }
+            }
+        }
+        contains_anonymous
+    };
```

Comment:
Why should this be a closure instead of a function?

Reply:
I have turned it into a function now, although I don't think it makes a difference.